lib/puppet/type/file/checksum.rb

```diff
@@ -15,6 +15,12 @@
     Puppet[:digest_algorithm].to_sym
   end
 
+  validate do |value|
+    if Puppet::Util::Platform.fips_enabled? && (value == :md5 || value == :md5lite)
```

Comment:
The test in lib/puppet/defaults.rb#970 is grep(/^md5/), should these tests be the same?

Reply:
Ah yeah good point. I updated defaults.rb to check for explicit checks instead of grepping with regexp

Reply:
LGTM.